doc/develop/tools/vscode.rst

```diff
@@ -0,0 +1,77 @@
+.. _vscode_ide:
+
+VS Code
+#######
+
+`Visual Studio Code`_ is a popular cross-platform IDE that supports C projects and has a rich set of
+extensions.
+
+This guide describes the process of setting up VS Code for Zephyr's
+:zephyr:code-sample:`blinky` sample in VS Code.
+
+The instructions have been tested on Linux. In terms of the VSCode workflow, the steps would be the
+same for macOS and Windows, but make sure to adjust the paths if needed.
+
+Get VS Code
+***********
+
+`Download VS Code`_ and install it.
+
+Install the required extensions through the `Extensions` marketplace in the left panel.
+Search for the `C/C++ Extension Pack`_ and install it.
+
+Initialize a new workspace
+**************************
+
+This guide gives details on how to configure the :zephyr:code-sample:`blinky`
+sample application, but the instructions would be similar for any Zephyr project and :ref:`workspace
+layout <west-workspaces>`.
+
+Before you start, make sure you have a working Zephyr development environment, as per the
+instructions in the :ref:`getting_started`.
+
+Open the project in VS Code
+***************************
+
+#. In VS Code, select :menuselection:`File --> Open Folder` from the main menu.
+
+#. Navigate to your Zephyr workspace (i.e.the :file:`zephyrproject` folder in your HOME directory if
+   you have followed the Getting Started instructions), then select
+   :file:`zephyr/samples/basic/blinky` or another sample project folder.
```

Comment:
This needs to be consistent with the build instructions that follow -- if you instruct the user to builder from `zephyrproject/zephyr` then that's what they should be opening in VS code. If they open the blinky folder, there won't be a `build/compile_commands.json` file in it since it's in `zephyrproject/zephyr`

Reply:
I didn't see that! Updating all paths to be coherent with $HOME/zephyrproject/zephyr/samples/basic/blinky.